File: typescript/queue.ts
//////////////////////////////
// Create a Queue Structure
// 6/29/20
// Add data to the end of the queue, information leaves the front of the queue
//////////////////////////////

////////////CLASS/////////////////////

class Queue {
    #data:Array<any> = []

    enter(data:any){
        return this.#data.push(data)
    }

    leave(){
        return this.#data.shift()
    }

    viewQueue(){
        console.log(this.#data)
    }
}

//////////Example Code///////

const lineRide = new Queue()

lineRide.enter("Susie")
lineRide.enter("Bobby")
lineRide.enter("Ralph")
lineRide.viewQueue()
lineRide.leave()
lineRide.viewQueue()
lineRide.leave()
lineRide.viewQueue()
lineRide.leave()
lineRide.viewQueue()

File: ruby/Queue.rb
##########################################
## Queue Data Structure
##########################################
## 06-29-20/6-30-2020
##########################################

class Queue 
    def initialize
        @que = []
    end

    def enter (data)
        return @que.push data
    end

    def leave
        return @que.shift
    end

    def getQue
        return @que
    end
end

########Example Code##########


LineRide = Queue.new
LineRide.enter(1)
LineRide.enter(2)
LineRide.enter(3)
puts LineRide.getQue
LineRide.leave
puts LineRide.getQue